### PR TITLE
Introduce transaction management layer

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
This PR ports over the existing sdk v1 transaction management and makes adjustments to use it in the new sdk.
Also, adds 2 examples on how to use the worker to fund multiple account and mint multiple tokens.

### Test Plan
pnpm test

### CODE REVIEW

File: examples/typescript/batch_funds.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * or get an execution validation error.
+ *
+ * The TransactionWorker constructor accepts
+ * @param aptosConfig - a config object
+ * @param sender - the sender account
+ * @param maxWaitTime - the max wait time to wait before restarting the local sequence number to the current on-chain state
+ * @param maximumInFlight - submit up to `maximumInFlight` transactions per account
+ * @param sleepTime - If `maximumInFlight` are in flight, wait `sleepTime` seconds before re-evaluating
+ *
+ * Read more about it here {@link https://aptos.dev/guides/transaction-management}
```
Comment (0xmaayan) [inline examples/typescript/batch_funds.ts:19]: will point to the new sdk v2 docs

File: .eslintrc.js [hunk 2]
```diff
@@ -27,6 +27,7 @@ module.exports = {
     "max-classes-per-file": ["error", 10],
     "import/prefer-default-export": "off",
     "object-curly-newline": "off",
+    "no-restricted-syntax": ["error", "ForInStatement", "LabeledStatement", "WithStatement"],
```
Comment (0xmaayan) [inline .eslintrc.js:30]: Replacing airbnb rule with following, to re-enable "ForOfStatement"

Comment (gregnazario) [reply .eslintrc.js:30]: The not allowing for loops was the most annoying

File: src/api/transactionSubmission.ts [hunk 3]
```diff
@@ -14,6 +20,42 @@ export class TransactionSubmission {
...
+   * @param args.sender The sender account to sign and submit the transaction
+   * @param args.data An array of transaction payloads
+   * @param args.options optional. Transaction generation configurations (excluding accountSequenceNumber)
+   *
+   * @return void. Throws if any error
+   */
+  async batchTransactionsForSingleAccount(args: {
+    sender: Account;
+    data: InputGenerateTransactionPayloadData[];
+    options?: Omit<InputGenerateTransactionOptions, "accountSequenceNumber">;
```
Comment (gregnazario) [inline src/api/transactionSubmission.ts:42]: May be interesting to be able to set this for each transaction, but let's leave that out for now.

File: examples/typescript/batch_mint.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+const APTOS_NETWORK: Network = NetworkToNetworkName[process.env.APTOS_NETWORK] || Network.DEVNET;
+
+const config = new AptosConfig({ network: APTOS_NETWORK });
+const aptos = new Aptos(config);
+const COLLECTION_NAME = "My batch collection!";
+const tokensToMint = 100;
+
+async function main() {
+  const sender: Account = Account.generate();
+  console.log(`sender ${sender.accountAddress.toString()}`);
```
Comment (gregnazario) [inline examples/typescript/batch_mint.ts:40]: Can drop the toString()

File: examples/typescript/batch_mint.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+  // fund sender accounts
+  const funds: Array<Promise<UserTransactionResponse>> = [];
+
+  funds.push(aptos.fundAccount({ accountAddress: sender.accountAddress.toStringWithoutPrefix(), amount: 10000000000 }));
```
Comment (gregnazario) [inline examples/typescript/batch_mint.ts:45]: similarly this should be able to be done without the `toString...` right?

**Review by gregnazario — APPROVED** [review]

**Pull request merged**